src/shared/components/query/CancerStudySelector.tsx

```diff
@@ -307,45 +313,104 @@ export default class CancerStudySelector extends React.Component<
+                                                                  })`
+                                                                : `Select all listed studies ${
+                                                                      shownStudies.length <
+                                                                      this.store
+                                                                          .cancerStudies
+                                                                          .result
+                                                                          .length
+                                                                          ? 'matching filter'
+                                                                          : ''
+                                                                  }  (${
+                                                                      shownStudies.length
+                                                                  })`}
+                                                        </strong>
+                                                    </label>
+                                                </Then>
+                                                <Else>
+                                                    <label>
+                                                        <input
+                                                            type="checkbox"
+                                                            data-test="selectAuthorizedStudies"
+                                                            style={{ top: -2 }}
+                                                            onClick={
+                                                                this.handlers
+                                                                    .onCheckAuthorizedFiltered
+                                                            }
+                                                            checked={
+                                                                shownAndSelectedStudies.length ===
+                                                                    shownAndAuthorizedStudies.length &&
+                                                                shownAndAuthorizedStudies.length >
+                                                                    0
+                                                            }
+                                                        />
+                                                        <strong>
+                                                            {shownAndSelectedStudies.length ===
+                                                                shownAndAuthorizedStudies.length &&
+                                                            shownAndAuthorizedStudies.length >
+                                                                0
```

Comment:
This condition shows twice, maybe we can create a constant for it and find a good name for it. e.g. `authorizedStudiesChecked`?

Reply:
new const called shownStudiesLengthstring has been created